pokedex/views.py

```diff
@@ -0,0 +1,136 @@
+            deck = Deck.objects.create(name=deck_name, user=request.user)
+            selected_cards = Card.objects.filter(id__in=selected_cards_ids)
+            deck.cards.set(selected_cards)
+            deck.save()
+
+            messages.success(request, 'wow goodjob')
+            return redirect('profile')
+        else:
+            messages.error(request, 'error')
+
+    return render(request, 'pokedex/create_deck.html', {
+        'page_obj': page_obj,
+        'search_query': search_query,
+    })
+
+def home(request):
+    return render(request, 'pokedex/home.html')
+
+def register(request):
+    if request.method == 'POST':
+        form = UserCreationForm(request.POST)
+        if form.is_valid():
+            user = form.save()
+            auth_login(request, user)
+            return redirect('home')
+    else:
+        form = UserCreationForm()
+    return render(request, 'registration/register.html', {'form': form})
+
+def view_deck(request, deck_id):
+    deck = get_object_or_404(Deck, id=deck_id)
+    return render(request, 'view_deck.html', {'deck': deck})
+
+@login_required
+def delete_deck(request, deck_id):
+    deck = get_object_or_404(Deck, id=deck_id)
+
+    if request.method == 'POST':
+        deck.delete()
+        return redirect('profile')
+
+    return render(request, 'delete_deck.html', {'deck': deck})
+
+def fetch_cards_from_api():
+    response = requests.get('https://api.pokemontcg.io/v2/cards/xy1-1')
+    if response.status_code == 200:
+        return response.json()
+    return []
+
+def update_cards():
+    cards_data = fetch_cards_from_api()
+    for card_data in cards_data:
+        Card.objects.update_or_create(
+            id=card_data['id'],
+            defaults={
+                'name': card_data['name'],
+            }
+        )
+
+def card_list(request):
+    cards = Card.objects.all()
+    return render(request, 'card_list.html', {'cards': cards})
+
+@login_required
+def profile_view(request):
+    user = request.user
+    decks = Deck.objects.filter(user=user)
+    return render(request, 'registration/profile.html', {
+        'decks': decks,
+    })
+
+@login_required
+def profile(request):
+    decks = Deck.objects.filter(user=request.user)
+    return render(request, 'registration/profile.html', {'decks': decks})
+
+@login_required
+def edit_deck(request, pk):
+    cards = Card.objects.all().order_by('name')
+    paginator = Paginator(cards, 10)
+    page_number = request.GET.get('page', 1)
+    page_obj = paginator.get_page(page_number)
+
+    if request.method == 'POST':
+        form = DeckForm(request.POST)
+        if form.is_valid():
+            deck = form.save(commit=False)
+            deck.user = request.user
+            deck.save()
+
+            selected_cards = request.POST.getlist('cards')
+            deck.cards.set(selected_cards)
+            deck.save()
+            return redirect('success')
+    else:
+        form = DeckForm()
```

Comment:
przekazujesz pusty formularz, przez to na stronie edycji decku nie wiadomo które karty są juz w decku. Przydałoby się przekazać informacje ktore karty są już w decku i je wyświetlać na stronie edycji decka już zaznaczone.